altair_pandas/_core.py

```diff
@@ -23,15 +27,17 @@ def __init__(self, data):
 
     def _preprocess_data(self, with_index=True):
         # TODO: do this without copy?
-        data = self._data.copy()
-        if not data.name:
-            data.name = 'value'
+        data = self._data
         if with_index:
             if isinstance(data.index, pd.MultiIndex):
+                data = data.copy()
                 data.index = pd.Index(
                     [str(i) for i in data.index], name=data.index.name)
-            return data.reset_index()
-        return data.to_frame()
+            data = data.reset_index()
+        else:
+            data = data.to_frame()
+        # Column names must all be strings.
+        return data.rename(_valid_column, axis=1)
```

Comment:
@jakevdp -- should we expliclty use `columns` instead `axis` --`data.rename(columns=_valid_column)`